packages/backend-lalr/data/HappyTemplate.hs

```diff
@@ -1,67 +1,39 @@
 -- $Id: GenericTemplate.hs,v 1.26 2005/01/14 14:47:22 simonmar Exp $
 
-#ifdef HAPPY_GHC
-#  if !defined(__GLASGOW_HASKELL__)
-#    error `HAPPY_GHC` is defined but this code isn't being built with GHC.
-#  endif
-#  define ILIT(n) n#
-#  define IBOX(n) (Happy_GHC_Exts.I# (n))
-#  define FAST_INT Happy_GHC_Exts.Int#
+#if !defined(__GLASGOW_HASKELL__)
+#  error This code isn't being built with GHC.
+#endif
+#define ILIT(n) n#
+#define IBOX(n) (Happy_GHC_Exts.I# (n))
+#define FAST_INT Happy_GHC_Exts.Int#
 -- Do not remove this comment. Required to fix CPP parsing when using GCC and a clang-compiled alex.
-#  if __GLASGOW_HASKELL__ > 706
-#    define LT(n,m) ((Happy_GHC_Exts.tagToEnum# (n Happy_GHC_Exts.<# m)) :: Prelude.Bool)
-#    define GTE(n,m) ((Happy_GHC_Exts.tagToEnum# (n Happy_GHC_Exts.>=# m)) :: Prelude.Bool)
-#    define EQ(n,m) ((Happy_GHC_Exts.tagToEnum# (n Happy_GHC_Exts.==# m)) :: Prelude.Bool)
-#  else
-#    define LT(n,m) (n Happy_GHC_Exts.<# m)
-#    define GTE(n,m) (n Happy_GHC_Exts.>=# m)
-#    define EQ(n,m) (n Happy_GHC_Exts.==# m)
-#  endif
-#  define PLUS(n,m) (n Happy_GHC_Exts.+# m)
-#  define MINUS(n,m) (n Happy_GHC_Exts.-# m)
-#  define TIMES(n,m) (n Happy_GHC_Exts.*# m)
-#  define NEGATE(n) (Happy_GHC_Exts.negateInt# (n))
-#  define IF_GHC(x) (x)
+#if __GLASGOW_HASKELL__ > 706
+#  define LT(n,m) ((Happy_GHC_Exts.tagToEnum# (n Happy_GHC_Exts.<# m)) :: Prelude.Bool)
+#  define GTE(n,m) ((Happy_GHC_Exts.tagToEnum# (n Happy_GHC_Exts.>=# m)) :: Prelude.Bool)
+#  define EQ(n,m) ((Happy_GHC_Exts.tagToEnum# (n Happy_GHC_Exts.==# m)) :: Prelude.Bool)
 #else
-#  define ILIT(n) (n)
-#  define IBOX(n) (n)
-#  define FAST_INT Prelude.Int
-#  define LT(n,m) (n Prelude.< m)
-#  define GTE(n,m) (n Prelude.>= m)
-#  define EQ(n,m) (n Prelude.== m)
-#  define PLUS(n,m) (n Prelude.+ m)
-#  define MINUS(n,m) (n Prelude.- m)
-#  define TIMES(n,m) (n Prelude.* m)
-#  define NEGATE(n) (Prelude.negate (n))
-#  define IF_GHC(x)
+#  define LT(n,m) (n Happy_GHC_Exts.<# m)
+#  define GTE(n,m) (n Happy_GHC_Exts.>=# m)
+#  define EQ(n,m) (n Happy_GHC_Exts.==# m)
 #endif
+#define PLUS(n,m) (n Happy_GHC_Exts.+# m)
+#define MINUS(n,m) (n Happy_GHC_Exts.-# m)
+#define TIMES(n,m) (n Happy_GHC_Exts.*# m)
+#define NEGATE(n) (Happy_GHC_Exts.negateInt# (n))
+#define IF_GHC(x) (x)
+
 
 data Happy_IntList = HappyCons FAST_INT Happy_IntList
 
-#if defined(HAPPY_ARRAY)
-#  define CONS(h,t) (HappyCons (h) (t))
-#else
-#  define CONS(h,t) ((h):(t))
-#endif
+#define CONS(h,t) (HappyCons (h) (t))
 
-#if defined(HAPPY_ARRAY)
-#  define ERROR_TOK ILIT(0)
-#  define DO_ACTION(state,i,tk,sts,stk) happyDoAction i tk state sts (stk)
-#  define HAPPYSTATE(i) (i)
-#  define GOTO(action) happyGoto
-#  define IF_ARRAYS(x) (x)
-#else
-#  define ERROR_TOK ILIT(1)
-#  define DO_ACTION(state,i,tk,sts,stk) state i i tk HAPPYSTATE(state) sts (stk)
-#  define HAPPYSTATE(i) (HappyState (i))
-#  define GOTO(action) action
-#  define IF_ARRAYS(x)
-#endif
+#define ERROR_TOK ILIT(0)
```

Comment:
(Do not inline. It would be better to define `errorTok :: Int#` at the top-level; alas, in GHC we can't do that at the moment (https://gitlab.haskell.org/ghc/ghc/-/issues/17521))